tests/cpp/test_gpu3.cpp

```diff
@@ -8202,6 +8202,53 @@ TEST_F(NVFuserTest, ReverseMerge) {
   ASSERT_TRUE(t0.equal(cg_outputs.at(0)));
 }
 
+TEST_F(NVFuserTest, preallocated_outputs) {
+  // Fusion definition inspired from NVFuserTest.AsyncCompilation_CUDA
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeSymbolicTensor(2);
+  TensorView* tv1 = makeSymbolicTensor(1);
+  TensorView* tv2 = makeSymbolicTensor(2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+
+  TensorView* tv3 = add(tv0, IrBuilder::create<Val>(1.0)); // Group 0
+  TensorView* tv4 = max(tv3, {0}); // Group 0
+  TensorView* tv5 = add(tv4, tv1); // Group 0
+  TensorView* tv6 = add(tv5, tv2); // Group 1
+
+  fusion->addOutput(tv6);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({8, 5}, options);
+  at::Tensor t1 = at::randn({5}, options);
+  at::Tensor t2 = at::randn({8, 5}, options);
+  at::Tensor t6 = at::empty({8, 5}, options);
+
+  std::vector<c10::IValue> aten_inputs = {t0, t1, t2};
+  std::vector<at::Tensor> preallocated_output = {t6};
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto outputs =
+      executor_cache.runFusionWithInputs(aten_inputs, preallocated_output);
+
+  EXPECT_TRUE(executor_cache.getMostRecentKernelRuntime()->isSegmented())
+      << "segmentation didn't happen";
```

Comment:
The following EXPECT_EQ subsumes this one.

Reply:
right, let me remove it